File: tmp_pdf/extract.py
import os
from pypdf import PdfReader

def extract():
    src_dir = 'd:/online/public/images'
    dest_dir = 'd:/online/tmp_pdf'
    
    files = [f for f in os.listdir(src_dir) if f.endswith('.pdf')]
    for file in files:
        try:
            reader = PdfReader(os.path.join(src_dir, file))
            text = ""
            for page in reader.pages:
                extracted = page.extract_text()
                if extracted:
                    text += extracted + "\n"
            with open(os.path.join(dest_dir, f"{file}.txt"), 'w', encoding='utf-8') as f:
                f.write(text)
            print(f"Extracted: {file} ({len(text)} chars)")
        except Exception as e:
            print(f"Error extracting {file}: {e}")

if __name__ == '__main__':
    extract()

File: tmp_pdf/extract.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extract() {
  const files = fs.readdirSync('d:/online/public/images').filter(f => f.endsWith('.pdf'));
  for (const file of files) {
    const dataBuffer = fs.readFileSync(`d:/online/public/images/${file}`);
    const data = await pdf(dataBuffer);
    fs.writeFileSync(`d:/online/tmp_pdf/${file}.txt`, data.text);
    console.log(`Extracted: ${file} (${data.text.length} chars)`);
  }
}
extract();
